### PR TITLE
Jmeixner

### PR DESCRIPTION
**Pull request by JakubMeixner**

### CODE REVIEW

File: configs/ConfigGridMaker.py
Comment (ggruszczynski) [inline configs/ConfigGridMaker.py:1]: hm, please create it as a new file `configs/ConfigGridMaker_ADE.py`

**Review by ggruszczynski — CHANGES_REQUESTED** [review]
hm, please create it as a new file `configs/ConfigGridMaker_ADE.py` to avoid conflicts in the upcoming future.

File: configs/ConfigGridMaker.py
Comment (JakubMeixner) [reply configs/ConfigGridMaker.py:1]: ok

File: numerical_solvers/solvers/SpectralTurbulenceGenerator.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def generate_turbulence(self, time: int):
         v = t.real(t.fft.ifft2(v_hat))
 
        
-        if self.std_dev[time] < 1E-14:
+        if self.std_dev< 1E-14:
```
Comment (ggruszczynski) [inline numerical_solvers/solvers/SpectralTurbulenceGenerator.py:158]: `self.std_dev[time]` shall be an array
https://github.com/IDEAS-NCBR-Computer-Graphics-Group/generative-inverse-heat-dissipation/blob/d7f954e9279613ebe89087cc385fe0d77580d3c7/configs/ffhq/res_128_grayscale/default_lbm_ade_ffhq_128_config.py#L92

File: numerical_solvers/solvers/LBM_SolverBase.py [hunk 2]
```diff
@@ -25,6 +25,10 @@ def __init__(self, domain_size, kin_visc, cs2,
         self.omega_kin.from_numpy(1.0 / (3.0* kin_visc + 0.5))
                 
         self.cs2 = ti.field(ti.f32, shape=self.max_iter[None])
+        if isinstance(cs2, float):
```
Comment (ggruszczynski) [inline numerical_solvers/solvers/LBM_SolverBase.py:28]: cs2 shall be an array, similarly to self.std_dev[time]

**Review by ggruszczynski — CHANGES_REQUESTED** [review]
cs2 and turb_intensity are arrays (filled with values accoridng to the chosen scheduler).
Ex
[turb_intensity](https://github.com/IDEAS-NCBR-Computer-Graphics-Group/generative-inverse-heat-dissipation/blob/d7f954e9279613ebe89087cc385fe0d77580d3c7/configs/ffhq/res_128_grayscale/default_lbm_ade_ffhq_128_config.py#L92)

File: numerical_solvers/solvers/LBM_SolverBase.py (hunk 2 above)
Comment (JakubMeixner) [reply numerical_solvers/solvers/LBM_SolverBase.py:28]: done

File: numerical_solvers/solvers/SpectralTurbulenceGenerator.py
[the remark this answers is not in the document]
Comment (JakubMeixner) [reply numerical_solvers/solvers/SpectralTurbulenceGenerator.py:1]: back [time]

File: scripts/datasets.py [hunk 3]
```diff
@@ -108,10 +108,10 @@ def get_dataset(config, uniform_dequantization=False, train_batch_size=None,
         test_data = datasets.LSUN(
             root="data/lsun", classes=['church_outdoor_val'], transform=transform)
     elif config.data.dataset == 'FFHQ':
-        trainloader = load_data(data_dir="data/ffhq-dataset/images1024x1024",
+        trainloader = load_data(data_dir="data/ffhq_dataset",
```
[the remark this answers is not in the document]
Comment (JakubMeixner) [reply scripts/datasets.py:111]: done

File: scripts/losses.py [hunk 4]
```diff
@@ -127,9 +127,9 @@ def get_step_lbm_fn(train, config,
     loss_fn = get_inverse_lbm_ns_loss_fn(train, config.model.sigma)
     
     # For automatic mixed precision
-    # scaler = torch.cuda.amp.GradScaler()
+    scaler = torch.cuda.amp.GradScaler()
     # scaler = torch.cuda.amp.GradScaler('cuda')
-    scaler = torch.amp.GradScaler("cuda")
```
[the remark this answers is not in the document]
Comment (JakubMeixner) [reply scripts/losses.py:132]: back

**Review by JakubMeixner — COMMENTED** [review]
back

File: numerical_solvers/corruptors/LBM_Base_Corruptor.py [hunk 5]
```diff
@@ -78,7 +78,7 @@ def _corrupt(self, x, steps, generate_pair=False):
             self._intermediate_samples[0][c] = torch.tensor( # rescale for preview
                 normalize_grayscale_image_range(np_gray_img, 0., 1.)).unsqueeze(0).clone()
 
-            for i in range(1, steps+1):
```
[the remark this answers is not in the document]
Comment (JakubMeixner) [reply numerical_solvers/corruptors/LBM_Base_Corruptor.py]: done

File: SLURM_batch_jobs/slurm_submitter_ideas_IHD.sh [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        echo "#SBATCH --time=$TIME"                  # Use the specified or default time
+        
+        echo
+
+        echo 'echo "Job started on $(date) at $HOSTNAME"'
+        echo "source \$HOME/py-ihd-env/bin/activate"
+        echo "cd \$HOME/generative-inverse-heat-dissipation"
+        
+        # Set the command to run with the specific config file
+        CMD="python train.py --config $CONFIG_FILE --workdir runs/ffhq/ffhq256_sigma_128"
```
[the remark this answers is not in the document]
Comment (JakubMeixner) [reply SLURM_batch_jobs/slurm_submitter_ideas_IHD.sh:80]: that is to be resolved i guess in the issue added by @mwlodarzc

**Comment by JakubMeixner** [comment]
@mwlodarzc thanks for PR and moreover for valuable comments!
